lib/rucio/client/ruleclient.py

```diff
@@ -30,7 +30,7 @@ class RuleClient(BaseClient):
 
     def add_replication_rule(
         self,
-        dids: list[str],
+        dids: list[dict],
```

Comment:
LGTM, but this would be more accurate:

```suggestion
        dids: Sequence[dict[str, str]],
```

Pyright will then likely complain about this code:

https://github.com/rucio/rucio/blob/339744866e9550f0f068f8e3c9574a95ccaa7d59/lib/rucio/api/rule.py#L92-L93

Because we're passing an `InternalScope` to something that should only be a `str`. 

To fix this, we can replace this `for d in dids` loop with this list comprehension:

```python
dids_with_internal_scope = [{'name': d['name'], 'scope': InternalScope(d['scope'], vo=vo)} for d in dids]
```

and then pass `dids=dids_with_internal_scope` to `rule.add_rule`.

(feel free to use a better name than `dids_with_internal_scope` - nothing better came to mind for me)

Reply:
`Sequence` is not imported in `ruleclient` at the moment, so you will need to import it from `collections.abc`:

```python
from collections.abc import Sequence
```